users/jdelkins/jdelkins.c

```diff
@@ -0,0 +1,286 @@
+
+        case BK_WRD:
+            do_mac_key(LCTL(KC_LEFT), ROPT(KC_LEFT), record);
+            break;
+
+        case KB_BOL:
+            do_mac_key(KC_HOME, RCMD(KC_LEFT), record);
+            break;
+
+        case KB_EOL:
+            do_mac_key(KC_END, RCMD(KC_RIGHT), record);
+            break;
+
+        case TG_SYS:
+            if (record->event.pressed) {
+                user_config.system_mac ^= 1;
+                eeconfig_update_user(user_config.raw);
+            }
+            break;
+
+        case KB_COPY:
+            do_mac_key(LCTL(KC_INS), RCMD(KC_C), record);
+            break;
+
+        case KB_PASTE:
+            do_mac_key(LSFT(KC_INS), RCMD(KC_V), record);
+            break;
+
+        case MY_GUI:
+            do_mac_key(KC_LGUI, KC_LOPT, record);
+            break;
+
+        case MY_ALT:
+            do_mac_key(KC_LALT, KC_LCMD, record);
+            break;
+
+        case MY_RGUI:
+            do_mac_key(KC_RGUI, KC_ROPT, record);
+            break;
+
+        case MY_RALT:
+            do_mac_key(KC_RALT, KC_RCMD, record);
+            break;
+
+        case MY_CALC:
+            do_mac_key(KC_CALC, KC_F14, record);
+            break;
+
+        case KB_MAKE:
+            if (!get_mods()) {
+                if (!record->event.pressed)
+                    SEND_STRING("make " QMK_KEYBOARD ":" QMK_KEYMAP SS_TAP(X_ENTER));
+                return false;
+            }
+            break;
+
+        case KB_VRSN:
+            if (!get_mods()) {
+                if (!record->event.pressed) {
+                    if (user_config.system_mac) {
+                        SEND_STRING(QMK_KEYBOARD "/" QMK_KEYMAP " @ " QMK_VERSION " (mac mode)");
+                    } else {
+                        SEND_STRING(QMK_KEYBOARD "/" QMK_KEYMAP " @ " QMK_VERSION " (non-mac mode)");
+                    }
+                }
+                return false;
+            }
+            break;
+
+        case KB_BOOT:
+            if (!get_mods()) {
+                if (record->event.pressed) {
+                    boot_timer = timer_read32();
+                } else {
+                    if (timer_elapsed32(boot_timer) >= 500) {
+                        reset_keyboard();
+                    }
+                }
+                return false;
+            }
+            break;
+
+        case KB_FLSH:
+            if (!get_mods()) {
+                if (!record->event.pressed) {
+                    SEND_STRING("make " QMK_KEYBOARD ":" QMK_KEYMAP
+#if defined(__ARM__) || defined(__arm__)
+                            ":dfu-util"
+#elif defined(BOOTLOADER_DFU) || defined(BOOTLOADER_LUFA_DFU) || defined(BOOTLOADER_QMK_DFU)
+                            ":dfu "
+#elif defined(BOOTLOADER_HALFKAY)
+                            ":teensy "
+#elif defined(BOOTLOADER_CATERINA)
+                            ":avrdude "
+#else
+                            ":doflash "
+#endif
+                            SS_TAP(X_ENTER));
```

Comment:
```suggestion
                    SEND_STRING("make " QMK_KEYBOARD ":" QMK_KEYMAP ":flash\n");
```

Reply:
Great suggestion. I have one keyboard (massdrop/alt) that doesn't support :flash so I still need to special-case that one.

Reply:
I think you should be able to set `PROGRAM_CMD` in rules.mk to allow `:flash` to run that if no other bootloader is specified, rather than defining your own flash target.

Reply:
Okay, that's a good idea. However, this requires a small change to `tmk_core/arm_atsam.mk` to make use of `PROGRAM_CMD` which I'll submit as a separate PR since it's outside of the keyboard tree